### ISSUE TITLE
Mixture model failure for +2. No issues with Comet/TPP of same data

### ISSUE THREAD

**Issue by jwinget** [Nesvilab/philosopher#64]
System info:
System OS: Windows 7, Architecture: AMD64
Java Info: 1.8.0_181, Java HotSpot(TM) 64-Bit Server VM, Oracle Corporation

Version info:
FragPipe version 9.4
MSFragger version 20190530
Philosopher version v1.4.6 (build 1563480020)
Python 3.5.1

For the TPP pipeline:
Comet 2018.01 rev. 4
TPP 5.2.0

I'm trying to compare Comet search with PEFF enabled against open search with MSFragger.
The input to both pipelines is a set of mzMLs from 8 fractions + an unfractionated run.
I noticed that MSFragger searches returned ~50% fewer proteins than Comet, and further inspection revealed that the +2 charge state mixture model is failing during the PeptideProphet step in the Philosopher pipeline, but it passes when using Comet/TPP. I'm using the default Open Search parameters for FragPipe.

The relevant commands are:
FragPipe: `peptideprophet --nonparam --expectscore --decoyprobs --masswidth 1000.0 --clevel -2 --decoy rev_ --database E:\dbase\2019-06-10-td-rev-UP000005640.fas --combine [bunch of pepXMLs]`

Comet/TPP: `PeptideProphetParser.exe interact.pep.xml ACCMASS PPM MINPROB=0 DECOYPROBS DECOY=DECOY NONPARAM EXPECTSCORE`

FWIW in the Comet/TPP pipeline the pepXML files are combined with InteractParser, not PeptideProphet.

I also noticed that there might be trouble with the mass calibration by MSFragger, as I see several errors in the peptideprophet section of the log like:
```
Unknown file type. No file loaded.[file_root].mzBIN_calibrated
WARNING: cannot open data file [file_root].mzBIN_calibrated in msms_run_summary tag... unrecognized extension .mzBIN_calibrated, trying .mzML ...
WARNING: CANNOT correct data file [file_root].mzML in msms_run_summary tag...
Unknown file type. No file loaded[file_root].mzBIN_calibrated
```

I tried re-running FragPipe with mass calibration turned off, but this didn't change the issue with +2 charge.

My guess is there is a bad flag set somewhere but I've gone through the GUI several times and can't find anything obvious. Thanks in advance for your help.

**Comment by fcyu** [comment]
Could you please tell us what the fasta file was used for Comet since you were using PEFF which was supposed to be different from the one used by MSFragger?

Also, could you please provide the command line for ProteinProphet?

The warning regarding failed to load the mzBIN_calibrated is fine. It won't affect your result.

Best,

Fengchao

**Comment by jwinget** [comment]
For Comet searches I used a PEFF file recently downloaded from NextProt. Decoys were added using decoyfastagenerator.pl in the TPP.

The command lines for ProteinProphet were:
FragPipe: `proteinprophet --maxppmdiff 2000000 MSfragger\4\interact.pep.xml MSfragger\2\interact.pep.xml MSfragger\3\interact.pep.xml MSfragger\1\interact.pep.xml`
Comet/TPP: `ProteinProphet.exe interact.iproph.pep.xml interact.iproph.prot.xml NORMPROTLEN IPROPHET MINPROB=0`

A couple of notes:
* I've got 4 groups set up in FragPipe and it looks like it's combining the ProtXMLs which is fine
* In the Comet/TPP pipeline I also run InterProphetParser in between PeptideProphet and ProteinProphet, hence the IPROPHET flag. I also run ProteinProphet separately on the 4 interact.iproph.pep.xml files/groups.

**Comment by prvst** [comment]
@jwinget, @fcyu 

If you noticed that the models failed for the +2 during PeptideProphet then I suggest starting by double checking the search parameters. This is usually an indication that one or more parameter might be wrong.

**Comment by fcyu** [comment]
A quick searching told me that there are 42410 proteins in the PEFF, while I guess there are around 20K+ protein in your `2019-06-10-td-rev-UP000005640.fas`. Will that possibly be the reason?

**Comment by prvst** [comment]
@fcyu a target-decoy, homo sapiens Swiss-Prot protein database will have approximately 40 k proteins as well.

**Comment by fcyu** [comment]
@prvst Sorry, I am talking about the number of target proteins.

**Comment by jwinget** [comment]
@fcyu is right. NextProt contains ~42k entries because it separates splicing isoforms into individual entries. I don't _think_ this is the issue, though, since this should impact protein inference more than fitting target/decoy models.

After carefully examining the search parameters, I realized I had Cleavage set to "ENZYMATIC" instead of "SEMI", with Missed Cleavages set to "1" for either case. **After switching to "SEMI" for Cleavage, the +2 mixture model was successfully fit.**

Sorry for the "false" bug report. Closing this but leaving it in case other folks are dull like me :)
